Data/ByteString.hs

```diff
@@ -727,15 +740,17 @@ scanr :: (Word8 -> Word8 -> Word8) -> Word8 -> ByteString -> ByteString
 scanr f v (BS fp len) = unsafeDupablePerformIO $ withForeignPtr fp $ \a ->
     create (len+1) $ \q -> do
         poke (q `plusPtr` len) v
-        scanr_ v (len-1) a q
+        go a q
```

Comment:
The haddock for `scanr` could be more explicit as to the order of the resulting elements.